### PR TITLE
Adding type checking within MockLoggers log_line function

### PR DESCRIPTION
**Pull request by rsarvaria**
A quick assertion to make sure that people are only attempting to log strings/text, and that attempting to log other other types isn't eaten by the mock.

### CODE REVIEW

**Comment by asottile** [comment]
The function also accepts `bytes`.

Also this may be considered a breaking change since it previously allowed stringlikes (though we did not use that)

**Comment by rsarvaria** [comment]
Is bytes covered by six.text_type? I can also add stringlikes to the check to make it non-breaking. This passed all tests on py26,py27,py34.

File: clog/loggers.py [hunk 1]
```diff
@@ -303,7 +303,7 @@ def __init__(self):
         self.lines = {}
 
     def log_line(self, stream, line):
-        assert isinstance(line, six.text_type)
+        assert isinstance(line, (bytes, six.string_types))
```
Comment (asottile) [inline clog/loggers.py]: This should be `(bytes, six.text_type)` (string_types includes `bytes` in python2)

Should you alos assert on the stream name too?

Double check what types are legal in python3.  I'm pretty sure it accepts bytes but we wouldn't want to get this wrong :)

Comment (rsarvaria) [inline clog/loggers.py]: Okay, I assume stream name can only be string?

Tried to switch tests/test_clog.py::TestFileLogger::test_log_line[hello-hello\nworld\n] to log a string casted to bytes, and it broke in py34... So I'm assuming that this means it's not a legal type.

Comment (asottile) [inline clog/loggers.py]: bytes are legal for both of them (tested this round trip):

```
(Pdb) clog.log_line('tmp_asottile', 'ohai')
(Pdb) clog.log_line('tmp_asottile', b'ohai')
(Pdb) clog.log_line(b'tmp_asottile', 'ohai')
(Pdb) clog.log_line(b'tmp_asottile', b'ohai')
(Pdb) import sys; sys.version_info
sys.version_info(major=3, minor=5, micro=1, releaselevel='final', serial=0)
```

File: clog/loggers.py [hunk 2]
```diff
@@ -303,6 +303,8 @@ def __init__(self):
         self.lines = {}
 
     def log_line(self, stream, line):
+        assert isinstance(stream, (bytes, six.text_type))
+        assert isinstance(line, (bytes, six.text_type))
```
Comment (asottile) [inline clog/loggers.py]: prefer 2 arg assertions so you get a useful message:

```python
# Sadness
>>> assert isinstance(x, (bytes, six.text_type))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
# Slightly better
>>> assert isinstance(x, (bytes, six.text_type)), x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: <object object at 0x7f1450a430a0>
# Perhaps best?
>>> assert isinstance(x, (bytes, six.text_type)), type(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: <type 'object'>
```

Comment (rsarvaria) [inline clog/loggers.py]: Yeah, the latter is probably best. Don't think people will be needing the hex for debugging purposes, and definitely looks cleaner

**Comment by asottile** [comment]
:boat:

**Pull request merged**